[ONNX] Non-global diagnostic context

BowenBao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #100407
* #100299
* __->__ #100219
* #99944
* #99668

Summary:

* `dynamo_export`, and everything within now access diagnostic context through a maintained local
  variable, instead of global.
* Refactored `diagnose_call` decorator to require local diagnostic context, instead of accessing global.
* Modified `test_fx_to_onnx_*.py` tests to only log '*.sarif' logs when `verbose=True`.
* Temporarily removed diagnostics for `OnnxFunction`, as they don't have access to diagnostic context
  anymore. These diagnostics will be the responsibility of `onnxscript`, and they will return once 
  diagnostics system is integrated there.

BowenBao committed:
[ONNX] Non-global diagnostic context

[ghstack-poisoned]

BowenBao committed:
Adjust unittest sarif logging on "[ONNX] Non-global diagnostic context"




[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Non-global diagnostic context"




[ghstack-poisoned]

BowenBao committed:
linter on "[ONNX] Non-global diagnostic context"




[ghstack-poisoned]

BowenBao committed:
linter on "[ONNX] Non-global diagnostic context"


Summary:

* `dynamo_export`, and everything within now access diagnostic context through a maintained local
  variable, instead of global.
* Refactored `diagnose_call` decorator to require local diagnostic context, instead of accessing global.
* Modified `test_fx_to_onnx_*.py` tests to only log '*.sarif' logs when `verbose=True`.
* Temporarily removed diagnostics for `OnnxFunction`, as they don't have access to diagnostic context
  anymore. These diagnostics will be the responsibility of `onnxscript`, and they will return once 
  diagnostics system is integrated there.


[ghstack-poisoned]

justinchuby commented on torch/onnx/_internal/fx/passes/fx_to_onnxscript.py:
```diff
@@ -7,13 +7,14 @@
 import onnxscript  # type: ignore[import]
 from onnxscript import evaluator, opset18  # type: ignore[import]
-from onnxscript.function_libs.torch_lib import graph_building  # type: ignore[import]
+from onnxscript.function_libs.torch_lib import (  # type: ignore[import]  # type: ignore[attr-defined]
```
```suggestion
from onnxscript.function_libs.torch_lib import (  # type: ignore[import,attr-defined]
```
Nit

justinchuby commented on torch/onnx/_internal/fx/passes/fx_to_onnxscript.py:
```diff
@@ -26,50 +27,12 @@
     fn: Callable, args: Tuple[Any, ...], kwargs: Dict[str, Any]
 ) -> str:
-    assert len(args) > 0
-    node = args[0]
```
Comment on 1 because it seems tricky?

BowenBao replied:
Sure. I was thinking in another PR to change the signature style along with the infra to 
```
def _fx_node_to_onnx_message_formatter(
    fn: Callable, diagnostic_context: DiagnosticContext, node: torch.fx.Node, *args, **kwargs
) -> str:
```
wdyt?

justinchuby replied:
Looking good!

BowenBao committed:
Update on "[ONNX] Non-global diagnostic context"


Summary:

* `dynamo_export`, and everything within now access diagnostic context through a maintained local
  variable, instead of global.
* Refactored `diagnose_call` decorator to require local diagnostic context, instead of accessing global.
* Modified `test_fx_to_onnx_*.py` tests to only log '*.sarif' logs when `verbose=True`.
* Temporarily removed diagnostics for `OnnxFunction`, as they don't have access to diagnostic context
  anymore. These diagnostics will be the responsibility of `onnxscript`, and they will return once 
  diagnostics system is integrated there.


[ghstack-poisoned]

BowenBao committed:
rebase on "[ONNX] Non-global diagnostic context"


Summary:

* `dynamo_export`, and everything within now access diagnostic context through a maintained local
  variable, instead of global.
* Refactored `diagnose_call` decorator to require local diagnostic context, instead of accessing global.
* Modified `test_fx_to_onnx_*.py` tests to only log '*.sarif' logs when `verbose=True`.
* Temporarily removed diagnostics for `OnnxFunction`, as they don't have access to diagnostic context
  anymore. These diagnostics will be the responsibility of `onnxscript`, and they will return once 
  diagnostics system is integrated there.


[ghstack-poisoned]

BowenBao commented:
@pytorchbot merge

BowenBao commented:
@pytorchbot merge -f "unrelated failure due to active SEV. ONNX tests passed on all three PRs."
